File: 8_fibonacci/fibonacci.jl
function fibonacci(n::Int)
    nums = [1, 1]
    while nums[end] < n
        push!(nums, nums[end] + nums[end-1])
    end

    for num in nums
        if num <=n
            println(num)
        end
    end
end

print("Enter an upper bound: ")
n = readline()
n = parse(Int, n)
fibonacci(n)

File: 8_fibonacci/fibonacci.cpp
#include <iostream>
#include <vector>

using namespace std;

void fibonacci(int n) {
    vector<int> nums = {1, 1};

    while (nums.back() < n) {
        int last_idx = nums.size() - 1;
        nums.push_back(nums[last_idx] + nums[last_idx - 1]);
    };

    for (int i = 0; i < nums.size(); i++) {
        if (nums[i] <= n) {
            cout << nums[i] << endl;
        };
    };
};

int main() {
    cout << "Enter an upper bound: ";
    int n;
    cin >> n;

    fibonacci(n);
};